### ISSUE TITLE
nix sign-paths as unprivileged user?

### ISSUE THREAD

**Issue by lheckemann** [NixOS/nix#1708]
```
% nix sign-paths ./result -k test.priv
error: you are not privileged to add signatures
```

I don't see a security reason for this not to be allowed for all `allowed-users`. Is there one? If not, could we allow this?

**Comment by edude03** [comment]
Running into this as well

**Comment by tonylotts** [comment]
I'm also hitting this on macOS.

**Comment by lheckemann** [comment]
Still valid

**Comment by codygman** [comment]
I'm blocked by this.

**Comment by lheckemann** [comment]
@codygman any chance you could implement this then? I'd be happy to give it a review.

**Comment by thufschmitt** [comment]
Discussed during today's Nix team meeting:

That makes sense. The only risk is non-trusted users ddosing the store (and eventually a binary cache) by adding a bazillion signatures to store paths. But that's an acceptable tradeoff given that allowed users already have the ability to DDos a Nix store (if only by creating an army of small paths).

Maybe we could mitigate the risk for the binary caches by making `nix copy` (and equivalent) only copy the signatures that can be verified, but that shouldn't be a blocker here.

PRs welcome :)

**Comment by Ericson2314** [comment]
Ideally, it should be safe for stores to hold all the things, even bogus things. With input addressing, we can't quite do this, because of uniqueness constraints / canonicity concerns. But with content addressing, we could imagine having all sorts of bogus realisations which are ignored when they aren't assigned with trusted keys. This also gets us closer to two mutually untrusting users using the same system.

**Comment by lheckemann** [comment]
Fixed in #8589
